power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -126,9 +126,12 @@ struct TransformerTapRegulatorCalcParam {
     IntS status{};
 };
 
-struct TransformerTapRegulatorOptimizerOutput {
-    IntS tap_pos{na_IntS};
+struct TransformerTapPosition {
+    Idx2D index;
+    IntS tap_position;
+    TransformerTapPosition(Idx2D _index, IntS _tap_position) : index(_index), tap_position(_tap_position) {}
```

Comment:
```suggestion
    Idx2D index{};
    IntS tap_position{};
```

otherwise it does not satisfy `DefaultConstructible` which is a (soft) requirement for `std::vector`